### PR TITLE
nixosTests: fix tests that relies on an empty rootfs using a bootloader

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

Some tests relies on booting on an empty rootfs to do their verification, this is not supported under `useBootLoader = true;` by default which expect you want a "full system image" with a proper Nix store.

Therefore, to make this work, we re-enable the host nix store.

A cleanup for overlay is included in this change, I will drop it.

Fixes the fallout for #227881.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
@ofborg test nixosTests.initrd-luks-empty-passphrase

**Comment by ElvishJerricco** [comment]
@RaitoBezarius Probably needs to hit `nixos/tests/lvm2/systemd-stage-1.nix` too, so that this PR can supersede https://github.com/NixOS/nixpkgs/pull/230366

**Comment by nikstur** [comment]
`useBootLoader` does not seems like a good abstraction. Maybe we should have `useDiskImage` to create a full disk image and `useBootLoader` to actually perform the installation of the boot loader and by default enable `useDiskImage`? Anyway just a thought, nothing actionable for this PR.

**Comment by RaitoBezarius** [comment]
@ofborg test initrd-luks-empty-passphrase

**Comment by RaitoBezarius** [comment]
> @RaitoBezarius Probably needs to hit `nixos/tests/lvm2/systemd-stage-1.nix` too, so that this PR can supersede #230366

Fixing it.

File: nixos/tests/initrd-luks-empty-passphrase.nix [hunk 1]
```diff
@@ -18,6 +18,12 @@ in {
       emptyDiskImages = [ 512 ];
       useBootLoader = true;
       useEFIBoot = true;
+      # This requires to have access
+      # to a host Nix store as
+      # the new root device is /dev/vdb
+      # an empty 512MiB drive, containing no Nix store.
+      mountHostNixStore = true;
+      # TODO: noNixStore = true;
```
Comment (nikstur) [inline nixos/tests/initrd-luks-empty-passphrase.nix:26]: This comment isn't really relevant here.

**Review by nikstur — APPROVED** [review]
LGTM

**Review by lheckemann — APPROVED** [review]
LGTM besides what @nikstur has already pointed out

**Comment by nikstur** [comment]
`nixos/tests/luks.nix` also needs this fix.

**Comment by RaitoBezarius** [comment]
> These tests also need this fix:
> 
> * `nixos/tests/luks.nix`
> * `nixos/tests/lvm2/systemd-stage-1.nix`

lvm2-sd1 is already working. Thinpool is broken but maintainers didn't get back to me, cc @ajs124.

LUKS is fixed now, this is ready.

**Comment by nikstur** [comment]
I'm quite certain that at least `nixosTests.lvm2.lvm-raid-sd-stage-1-linux-latest` needs the `mountHostNixStore` fix. Just tested it again on `nixos-unstable`.

**Comment by RaitoBezarius** [comment]
@ofborg test nixosTests.lvm2.lvm-raid-sd-stage-1-linux-latest

**Comment by RaitoBezarius** [comment]
@ofborg test lvm2.lvm-raid-sd-stage-1-linux-latest

**Comment by nikstur** [comment]
> 



> > I'm quite certain that at least `nixosTests.lvm2.lvm-raid-sd-stage-1-linux-latest` needs the `mountHostNixStore` fix. Just tested it again on `nixos-unstable`.
> 
> which is here: [ff3936e](https://github.com/NixOS/nixpkgs/commit/ff3936e14b727d3a29d1a3212d6919055ac61804) right ?

Ah sorry, somehow, overlooked this multiple times.

**Review by nikstur — APPROVED** [review]

**Pull request merged**